### ISSUE TITLE
Function alternative to the `eof!` macro

### ISSUE THREAD

**Issue by shnewto** [Geal/nom#1061]
While updating the [bnf](https://github.com/shnewto/bnf) crate from macros (nom 3.2) to nom 5.1 functions recently, I came up against some logic that used the `eof!` macro. I couldn't find a 1 to 1 function to replace it with and ended up implementing my own. 

Here's what I ended up with:
```rust
pub fn eof<I: Copy + InputLength, E: ParseError<I>>(input: I) -> IResult<I, I, E> {
    if input.input_len() == 0 {
        Ok((input, input))
    } else {
        Err(nom::Err::Error(E::from_error_kind(input, ErrorKind::Eof)))
    }
}

```

I'd be happy to make a PR to add something like this but there may have been a good reason to omit a `pub fn eof` and so I thought I'd check before jumping into anything.

So, I suppose. If there's not going to be a 1 to 1 `pub fn eof` function, what would be a good alternative?

If there could be a `pub fn eof`, would you be open to a PR?

**Comment by mdorman** [comment]
Since I just borrowed this for a project, I think you should just open the PR for it; it seems like an oversight that it wasn't converted.

**Comment by Geal** [comment]
fixed in #1112
